In [1]:
from sogaPreprocessor import *
from producecfg import *
from smoothcfg import *
from libSOGA import *

torch.set_default_dtype(torch.float64)

# Old Experiments

In [7]:
#program = 'AQUA/TimeSeries'
#file = '../programs/SOGA/' + program + '.soga'
#compiled = compile2SOGA(file)
#cfg = produce_cfg(compiled)
#smooth_cfg(cfg)
#output = start_SOGA(cfg)
#output.gm.mean()
#print(output.var_list)
#print(output.gm.mean())
#cfg.node_list

In [8]:
import time

programs = ['BayesPointMachine',   # OK
            'Bernoulli',           # OK
            'Burglar',             # OK 
            # 'ClickGraph',          # too many components (> 1 million)
            'ClinicalTrial',       # OK 
            'CoinBias',            # OK 
            #'DigitRecognition', 
            'Grass',               # OK 
            #'IndianGPA',         
            'MurderMistery',       # OK
            'NoisyOr',             # OK 
            'SurveyUnbias',        # OK
            'ThreeCoins',          # OK 
            'TrueSkills',          # OK
            'TwoCoins',            # OK 
            'AQUA/Altermu',        # OK
            'AQUA/Altermu2',       # OK
            #'AQUA/NormalMixturesPrune',
            'AQUA/RadarQuery',     # OK
            'AQUA/TimeSeries'      # OK
            ]

for program in programs:
    file = '../programs/SOGA/' + program + '.soga'
    compiled = compile2SOGA(file)
    cfg = produce_cfg(compiled)
    smooth_cfg(cfg, smooth_eps=1e-5)
    start = time.time()
    output = start_SOGA(cfg)
    end = time.time()
    print()
    print(program + '| time: {} | mean: {} | var_list: '.format(np.round(end-start,2), output.gm.mean()) + str(output.var_list))


BayesPointMachine| time: 0.07 | mean: tensor([ 0.0112, -0.1202, -0.2778, -4.1396, -2.8644, -4.7678, -2.9071, -2.1989,
        -4.8623]) | var_list: ['w[0]', 'w[1]', 'w[2]', 'mean[0]', 'mean[1]', 'mean[2]', 'mean[3]', 'mean[4]', 'mean[5]']

Bernoulli| time: 0.16 | mean: tensor([0.2519, 1.0000]) | var_list: ['theta', 'y']

Burglar| time: 0.1 | mean: tensor([2.8221e-04, 2.9934e-03, 3.2788e-03, 1.0000e+00, 1.0000e+00, 1.0000e+00]) | var_list: ['earthquake', 'burglary', 'alarm', 'phoneWorking', 'maryWakes', 'called']

ClinicalTrial| time: 0.27 | mean: tensor([0.7531, 0.6606, 0.3399, 0.0000, 1.0000]) | var_list: ['isEff', 'probTreated', 'probContr', 'ycontr', 'ytreat']

CoinBias| time: 0.11 | mean: tensor([0.4247, 0.0000]) | var_list: ['bias', 'y']

Grass| time: 0.22 | mean: tensor([0.5758, 0.7079, 0.4298, 0.7000, 0.4955, 0.9583, 0.9305, 0.6954, 0.4172,
        1.0000]) | var_list: ['cloudy', 'rain', 'sprinkler', 'temp1', 'wetRoof', 'temp2', 'temp3', 'or1', 'or2', 'wetGrass']

MurderMistery


BayesPointMachine| time: 0.18 | mean: [ 0.01117781 -0.12016474 -0.27778036 -4.13952624 -2.86434177 -4.76773406
 -2.90703202 -2.19885417 -4.86233859] | var_list: ['w[0]', 'w[1]', 'w[2]', 'mean[0]', 'mean[1]', 'mean[2]', 'mean[3]', 'mean[4]', 'mean[5]']

Bernoulli| time: 21.63 | mean: [0.2568569 1.       ] | var_list: ['theta', 'y']

Burglar| time: 0.04 | mean: [2.82212318e-04 2.99344924e-03 3.27537935e-03 1.00000000e+00
 1.00000000e+00 1.00000000e+00] | var_list: ['earthquake', 'burglary', 'alarm', 'phoneWorking', 'maryWakes', 'called']

ClinicalTrial| time: 47.36 | mean: [0.75341998 0.65794587 0.34258052 0.         1.        ] | var_list: ['isEff', 'probTreated', 'probContr', 'ycontr', 'ytreat']

CoinBias| time: 0.56 | mean: [0.42600849 0.        ] | var_list: ['bias', 'y']

Grass| time: 0.07 | mean: [0.57579972 0.70792768 0.42976356 0.7        0.49554937 0.95827538
 0.93045897 0.69541029 0.41724618 1.        ] | var_list: ['cloudy', 'rain', 'sprinkler', 'temp1', 'wetRoof', 'temp2', 'temp3', 'or1', 'or2', 'wetGrass']

MurderMistery| time: 0.01 | mean: [0.01581722 1.        ] | var_list: ['aliceDunnit', 'withGun']

NoisyOr| time: 0.2 | mean: [0.5        0.5        0.45       0.45       0.45       0.45
 0.6975     0.415      0.58825    0.81441875] | var_list: ['n0', 'n4', 'n1', 'n21', 'n22', 'n33', 'n2', 'n31', 'n32', 'n3']

SurveyUnbias| time: 1.35 | mean: [0.79946436 0.50043923 1.         1.        ] | var_list: ['bias1', 'bias2', 'ansb1', 'ansb2']

ThreeCoins| time: 0.01 | mean: [0.75 0.75 0.75] | var_list: ['a', 'b', 'c']

TrueSkills| time: 0.02 | mean: [104.7131673   97.60989202  97.67694068 115.31779373  92.23214907
  92.4500572 ] | var_list: ['skillA', 'skillB', 'skillC', 'perfA', 'perfB', 'perfC']

TwoCoins| time: 0.02 | mean: [0.33333333 0.33333333 0.        ] | var_list: ['first', 'second', 'both']

AQUA/Altermu| time: 0.27 | mean: [-2.57251482e+00  3.38062060e-01  2.71757796e+00  1.09861336e+00
  2.85603752e+00 -9.16513510e-01  1.55551270e-01 -2.68160347e+00
  2.47043789e+00  3.47459025e+00  1.63949862e+00 -1.32148757e+00
  2.64187513e+00  3.03578480e-01 -4.09546231e+00 -1.50709863e+00
 -9.95178660e-01 -2.06488920e+00 -2.40317949e+00  3.46383544e+00
  9.11736960e-01  1.18222221e+00  4.23572200e-02 -5.28151710e-01
  1.15551598e+00 -1.62749724e+00  7.14732370e-01 -1.08458812e+00
  4.66020296e+00  1.24563831e+00 -6.79708620e-01  9.34616810e-01
  1.18187607e+00 -1.49501051e+00  2.44755622e+00 -2.06424237e+00
 -4.58407400e-02  1.93396696e+00  1.07685273e+00 -9.83790700e-02
  0.00000000e+00  0.00000000e+00 -1.37893048e-03  3.11638290e-01] | var_list: ['y[0]', 'y[1]', 'y[2]', 'y[3]', 'y[4]', 'y[5]', 'y[6]', 'y[7]', 'y[8]', 'y[9]', 'y[10]', 'y[11]', 'y[12]', 'y[13]', 'y[14]', 'y[15]', 'y[16]', 'y[17]', 'y[18]', 'y[19]', 'y[20]', 'y[21]', 'y[22]', 'y[23]', 'y[24]', 'y[25]', 'y[26]', 'y[27]', 'y[28]', 'y[29]', 'y[30]', 'y[31]', 'y[32]', 'y[33]', 'y[34]', 'y[35]', 'y[36]', 'y[37]', 'y[38]', 'y[39]', 'w1', 'w2', 'w3', 'mean']

AQUA/Altermu2| time: 0.19 | mean: [-2.57251482  0.33806206  2.71757796  1.09861336  2.85603752 -0.91651351
  0.15555127 -2.68160347  2.47043789  3.47459025  1.63949862 -1.32148757
  2.64187513  0.30357848 -4.09546231 -1.50709863 -0.99517866 -2.0648892
 -2.40317949  3.46383544  0.91173696  1.18222221  0.04235722 -0.52815171
  1.15551598 -1.62749724  0.71473237 -1.08458812  4.66020296  1.24563831
 -0.67970862  0.93461681  1.18187607 -1.49501051  2.44755622 -2.06424237
 -0.04584074  1.93396696  1.07685273 -0.09837907  0.1556079   0.1556079 ] | var_list: ['y[0]', 'y[1]', 'y[2]', 'y[3]', 'y[4]', 'y[5]', 'y[6]', 'y[7]', 'y[8]', 'y[9]', 'y[10]', 'y[11]', 'y[12]', 'y[13]', 'y[14]', 'y[15]', 'y[16]', 'y[17]', 'y[18]', 'y[19]', 'y[20]', 'y[21]', 'y[22]', 'y[23]', 'y[24]', 'y[25]', 'y[26]', 'y[27]', 'y[28]', 'y[29]', 'y[30]', 'y[31]', 'y[32]', 'y[33]', 'y[34]', 'y[35]', 'y[36]', 'y[37]', 'y[38]', 'y[39]', 'w1', 'w2']

AQUA/RadarQuery| time: 17.48 | mean: [ 0.2         0.36        5.19961521  5.19961521 -1.65956536  1.18160201
  0.6199988   5.          4.72166135] | var_list: ['b0', 'b1', 'x0', 'x1', 'x', 'y', 'flip', 'o0', 'o1']

AQUA/TimeSeries| time: 0.43 | mean: [-1.60828345  0.70074326  0.44734169 -1.09070023 -2.49513229 -2.07918899
 -3.34621083 -3.21905762 -3.8483514  -4.1459158  -2.17085703 -2.7298601
 -2.35417404 -3.05798313 -2.85360404 -2.9261493  -2.67713445 -2.97731975
 -4.75105001 -3.29399034 -4.11626048 -4.08086402 -3.34743298 -3.5929497
 -1.7048937  -3.52291184 -4.13944649 -4.04992281 -3.71707286 -2.68302052
 -3.21920266 -1.78221835 -1.71385612 -3.15284682 -3.26094915 -2.92240028
 -2.40186353 -3.16973318 -4.00540748 -2.32625323 -1.74827635 -1.05716215
 -3.44536781] | var_list: ['a', 'b', 'lamb', 'y0', 'y1', 'y2', 'y3', 'y4', 'y5', 'y6', 'y7', 'y8', 'y9', 'y10', 'y11', 'y12', 'y13', 'y14', 'y15', 'y16', 'y17', 'y18', 'y19', 'y20', 'y21', 'y22', 'y23', 'y24', 'y25', 'y26', 'y27', 'y28', 'y29', 'y30', 'y31', 'y32', 'y33', 'y34', 'y35', 'y36', 'y37', 'y38', 'y39']